deserialization.go

```diff
@@ -225,8 +194,12 @@ func transform[B any](ctx context.Context, body B) (B, error) {
 		err := inTransformerBody.InTransform(ctx)
 		if err != nil {
 			return body, BadRequestError{
+				Title:  "Transformation Failed",
 				Err:    err,
 				Detail: "cannot transform request body: " + err.Error(),
+				Errors: []ErrorItem{
+					{Name: "transformation", Reason: "transformation failed"},
+				},
 			}
 		}
```

Comment:
No tests added for this?

Reply:
You're right, I'll add some

Reply:
I'll work on another PR specially for error handling in the Transform, I think current implementation isn't ideal anyway